core/tests/bank_forks.rs

```diff
@@ -358,9 +359,9 @@ mod tests {
                     .set_root(current_bank.slot(), &snapshot_sender);
             }
 
-            let num_old_slots = num_set_roots * add_root_interval - MAX_CACHE_ENTRIES + 1;
+            let num_old_slots = num_set_roots * *add_root_interval - MAX_CACHE_ENTRIES + 1;
             let expected_slots_to_snapshot = (num_old_slots as u64
-                ..=num_set_roots as u64 * add_root_interval as u64)
+                ..=num_set_roots as u64 * *add_root_interval as u64)
```

Comment:
nits: The original code is a bit hard to read for me. This could be written like this:

```rust
let num_slots = num_set_roots * *add_root_interval;
let num_old_slots = num_slots - MAX_CACHE_ENTRIES + 1;
let expected_slots_to_snapshot = (num_old_slots as u64..=num_slots as u64).collect_vec();
```